### ISSUE TITLE
Calling PushPageModel outside of a Command doesn't push the Page

### ISSUE THREAD

**Issue by realies** [rid00z/FreshMvvm#171]
When calling `CoreMethods.PushPageModel` outside of a `Command` the requested PageModel is initialised but the view stays unchanged. In my case I have a SplashPageModel that is going to setup a service and connect to a server. To emulate the delay I've inserted a Task.Run in the constructor with a few Task.Delay instances to emulate delays. Here's what I mean:

```
public SplashScreenPageModel(IDataService dataService)
{
	this.dataService = dataService;
	Task.Run(async () => {
		this.Status = "Loading app...";
		await Task.Delay(TimeSpan.FromSeconds(2));
		this.Status = "Loading profile...";
		await Task.Delay(TimeSpan.FromSeconds(2));
		await CoreMethods.PushPageModel<CustomerHomePageModel>();
	});
}
```

The navigation service is defined as:
```
var splashScreen = FreshMvvm.FreshPageModelResolver.ResolvePageModel<SplashScreenPageModel>();
var navContainer = new FreshMvvm.FreshNavigationContainer(splashScreen);
MainPage = splashScreen;
```

The constructor of `CustomerHomePageModel` has `Debug.WriteLine("CustomerHomePageModel Initialised");` which gets called, but the visible page does not change. What would be an appropriate approach for the above? I have tried calling the method from an event handler with the same result.

**Comment by EmilAlipiev** [comment]
I think that you should use init or viewisappearing Methods. Constructors is called before init if i am not mistaken thats why pushpagemodel doesnt work.

**Comment by realies** [comment]
@EmilAlipiev, there's an awaited Task.Delay that delays for long enough for everything to be initialised.

**Comment by EmilAlipiev** [comment]
No, what I meant is that in order to push a page from a pagemodel (viewmodel), init method needs to be executed and constructor is called before init. for that reason, pushpagemodel might not be working. I am not sure but just give it a try in init or viewappearing methods, both can be async as well

**Comment by realies** [comment]
@EmilAlipiev, can you refer your statement to the documentation or an official example?
